tests/test_live.py

```diff
@@ -0,0 +1,1379 @@
+
+                if not local_dir.exists():
+                    local_dir = local_path / 'local-dir'
+                    local_dir.mkdir()
+
+                self.assertRaises(
+                    subprocess.CalledProcessError,
+                    call_gsutil_cp,
+                    src=test_case[0],
+                    dst=test_case[1],
+                    recursive=True)
+                if pathlib.Path(test_case[1]).is_dir():
+                    shutil.rmtree(test_case[1], True)
+
+    def test_gsutil_vs_gswrap_download_non_recursive(self):  # pylint: disable=invalid-name
+        # pylint: disable=too-many-locals
+        with temppathlib.TemporaryDirectory() as tmp_dir:
+            local_path = tmp_dir.path / 'tmp'
+            local_path.mkdir()
+            local_file = local_path / 'local-file'
+            local_file.write_text('hello')
+            another_local_file = local_path / 'another-local-file'
+            another_local_file.write_text('hello again')
+            local_dir = local_path / 'local-dir'
+            local_dir.mkdir()
+
+            # yapf: disable
+            test_cases = [
+                ["gs://{}/{}/d1/f11".format(TEST_GCS_BUCKET, self.bucket_prefix)
+                    , (local_path / 'uninitialized-file').as_posix()],
+                ["gs://{}/{}/d1/f11".format(TEST_GCS_BUCKET, self.bucket_prefix)
+                    , local_file.as_posix()],
+            ]
+            # yapf: enable
+
+            gsutil_ls_set = set()
+            gcs_ls_set = set()
+
+            ls_path = tmp_dir.path.as_posix()
+
+            for test_case in test_cases:
+                if not local_dir.exists():
+                    local_dir = local_path / 'local-dir'
+                    local_dir.mkdir()
+
+                self.client.cp(
+                    src=test_case[0], dst=test_case[1], recursive=False)
+
+                gcs_paths = ls_local(path=ls_path)
+                [gcs_ls_set.add(path) for path in gcs_paths]
+                if pathlib.Path(test_case[1]).is_dir():
+                    shutil.rmtree(test_case[1], True)
+
+                if not local_dir.exists():
+                    local_dir = local_path / 'local-dir'
+                    local_dir.mkdir()
+
+                call_gsutil_cp(
+                    src=test_case[0], dst=test_case[1], recursive=False)
+                gsutil_paths = ls_local(path=ls_path)
+                [gsutil_ls_set.add(path) for path in gsutil_paths]
+                if pathlib.Path(test_case[1]).is_dir():
+                    shutil.rmtree(test_case[1], True)
+
+                self.assertListEqual(gsutil_paths, gcs_paths)
+
+            self.assertListEqual(list(gsutil_ls_set), list(gcs_ls_set))
+
+    def test_gsutil_vs_gswrap_download_non_recursive_check_raises(self):  # pylint: disable=invalid-name
+        # pylint: disable=too-many-locals
+        with temppathlib.TemporaryDirectory() as tmp_dir:
+            local_path = tmp_dir.path / 'tmp'
```

Comment:
That would have been great, unfortunately I couldn't find any way with Marko's temppathlib. https://pypi.org/project/temppathlib/. It might be possible with the tempfile library below. Should I try this?